module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordDataFormatterTest.php

```diff
@@ -152,7 +152,7 @@ protected function getFormatter()
             new \VuFind\Config\PluginManager($container)
         );
         $this->addPathResolverToContainer($container);
-        $formatter = $factory($container, RecordDataFormatter::class);
+        $formatter = $factory($container, RecordDataFormatter::class)($this->getDriver());
```

Comment:
In order to exercise both forms of usage, maybe we can use a `@dataProvider` that passes in the name of a method that receives a driver and spec and returns the data. Then we can run the test twice, once for legacy invocation and once for new invocation.

If you're not sure what I'm talking about, I can probably find some time next week to put together a clearer example (I'm off for the rest of this week).

Reply:
Yeah, been doing something with the tests so I can try out and see it comes close to your thoughts!